bin/logos.pl

```diff
@@ -643,6 +645,8 @@ package main;
 			$property->class($currentClass);
 			$property->type($type);
 			$property->name($name);
+			$property->readonly($readonly);
```

Comment:
Can't place my comment in the area of L630-640 to request adding checks for cases like having both `readonly` and `retain` on the attribute list, which seems contradictory.

Reply:
Added this:

```perl
if($readonly) {
    if($retain || $copy || $assign) {
        fileError($lineno, "property attribute 'readonly' cannot be used with memory management attributes (assign, retain, copy).");
    }
}
```